client/client.go

```diff
@@ -72,6 +73,10 @@ func Configure(logger hclog.Logger, config interface{}) (schema.ClientMeta, diag
 	logger.Info("Trying to authenticate via CLI (azidentity)")
 	var azCred azcore.TokenCredential
 	azCred, err = azidentity.NewAzureCLICredential(nil)
+	if err == nil {
+		// check token generation
+		_, err = azCred.GetToken(context.Background(), policy.TokenRequestOptions{Scopes: []string{"https://management.core.windows.net//.default"}})
```

Comment:
@amanenk Does this fix the principal auth issue, or only report the error to the user earlier? Looking at [this doc](https://docs.microsoft.com/en-us/azure/developer/go/azure-sdk-authentication-service-principal?tabs=azure-cli), it looks like we may want to try using `DefaultAzureCredential`. Maybe we should even put it in a `ChainedTokenCredential` to fit with Azure's SDK conventions

Reply:
Ah, I'm reading the error message from the original issue more carefully now, and it suggests that the issue was that the `az` CLI wasn't found on the path. So what you're doing here is checking that we can auth using that method, and if not, falling through to environment variables. I see! I still think we may want to use a `ChainedTokenCredential` instead, so that the Azure SDK will go through all the auth options for us if one fails.

Reply:
@amanenk I also made an attempted fix here, using the default credential chain: https://github.com/cloudquery/cloudquery/pull/1283 

Let's chat about how we want to proceed: perhaps we can release your fix as a release here (the final one to be released on this repo), and the default credential one only on the monorepo, since that's also a breaking change.